README.md
Overview
========

To create a GeoSHAPE VM, you can clone this repo and simply do ```vagrant up``` please see the [How To Get GeoSHAPE] (https://docs.google.com/document/d/1KMpk6dXuqvwfEi0pfRpaGY62j6ikoYtpYUPU0sJQAmk/) document before creatign the VM as it has a lot of relevant information. Please visit www.geoshape.org for additional information. 


CHANGELOG.md
Change Log
==========
All notable changes to this project will be documented in this file.

[Unreleased][unreleased]
------------------------

[1.4] - 2015-05-22
------------------
- Added place holder for fqdn and variable to enable/disable ssl in dna.json
- Added geoshape-provision script to make it easy to apply dna.json changes
- Bump the Vagrant file to pull rogue-chef-repo release-1.4 by default
- Brings in updated maploom, geonode, and other fixes. 

[1.3] - 2015-04-08
------------------
- Bump the Berkshelf gem to prevent dependency conflict.

[1.2] - 2015-01-20
------------------
- Adds CHANGELOG.md.
- Bump the rogue-cookbook dependency to [release-1.2].

[1.1] - 2015-01-19
------------------
- Bump the rogue-cookbook dependency to [release-1.1].

[1.0] - 2015-01-15
------------------
- Initial ROGUE release re-released using 1.0 tag.


[unreleased]: https://github.com/ROGUE-JCTD/rogue-chef-repo/compare/release-1.4...HEAD
[1.4]: https://github.com/ROGUE-JCTD/rogue-chef-repo/compare/release-1.3...release-1.4
[1.3]: https://github.com/ROGUE-JCTD/rogue-chef-repo/compare/release-1.2...release-1.3
[1.2]: https://github.com/ROGUE-JCTD/rogue-chef-repo/compare/release-1.1...release-1.2
[1.1]: https://github.com/ROGUE-JCTD/rogue-chef-repo/compare/release-1.0...release-1.1
[1.0]: https://github.com/ROGUE-JCTD/rogue-chef-repo/tree/release-1.0


[release-1.2]: https://github.com/ROGUE-JCTD/rogue-cookbook/releases/tag/release-1.2
[release-1.1]: https://github.com/ROGUE-JCTD/rogue-cookbook/releases/tag/release-1.1


solo/geoshape-upgrade.sh
#!/bin/bash
# exit if anything returns failure
set -e

if [ "$1" = "silent" ];
then
  GEOSHAPE_SILENT=true
  GEOSHAPE_VERSION="$2"
elif [ "$2" = "silent" ];
then
  GEOSHAPE_SILENT=true
  GEOSHAPE_VERSION="$1"
else
  GEOSHAPE_VERSION="$1"
fi

if [ "$GEOSHAPE_SILENT" != true ]; then
    echo "====[ WARNING:
      You are attempting to upgrade the installation of geoshape. Before proceeding, you should backup
      this machine. If it is a virtual machine, you should create a snapshot so that you can undo the changes
      if there are any incompatibilities. In some cases, the database may no longer operate as expented resulting
      in potential loss of data. Note that only some upgrades can complete successfully! Before continuing
      you should review changes introduced between the geoshape version you are currently on and the version
      you are attempting to upgrade to. There could be many backwards incompatible changes introduced at
      any of the geoshape dependencies. In some cases, a clean installation of geoshape might be necessary
      to get the latest functionality successfully. For example, the geoserver_data directory required by
      the latest version of geoshape, may contain some configuration or scripts that doesn't make it
      possible to simply keep your current /var/lib/geoserver_data directory. Similar issues could arise for
      many other dependencies of geoshape. This script is only intended for advanced users! The safest way
      the to upgrade geoshape is to create a new instance and migrate your data."

    while true; do
        read -p "=> Are you sure you want to continue? " yn
        case $yn in
            [Yy]* ) break;;
            [Nn]* ) echo "    geoshape upgrade exited. ";exit;;
            * ) echo "    Please answer yes or no.";;
        esac
    done
fi

# store current directory
pushd .

cd /opt/rogue-chef-repo

# get latest tags/branchs
git fetch

if [ -z "$GEOSHAPE_VERSION" ];
then
  # discover the latest release tag
  RELEASE_TAGS=(`git tag`)
  echo 'release tags: '
  echo "${RELEASE_TAGS[@]}"
  # sort the list of branches that had '.' in them such that index 0 is the largest one
  RELEASE_TAGS_SORTED=($(printf '%s\n' "${RELEASE_TAGS[@]}"|sort -r))
  UPGRADE_TO_GEOSHAPE_VERSION=${RELEASE_TAGS_SORTED[0]}
fi

if [ -z "$GEOSHAPE_VERSION" ];
then
    echo 'Note: geoshape version was not specified, did you intend to upgrade to release: '${UPGRADE_TO_GEOSHAPE_VERSION}
    while true; do
        read -p "=> Are you sure you want to continue? " yn
        case $yn in
            [Yy]* ) break;;
            [Nn]* ) echo "    geoshape upgrade exited. ";exit;;
            * ) echo "    Please answer yes or no.";;
        esac
    done
fi

GEOSHAPE_VERSION=$UPGRADE_TO_GEOSHAPE_VERSION

# checkout the requests 'version' (tag, branch, commit)
git checkout ${GEOSHAPE_VERSION}

# if on a branch, do a pull to get latest on local branch. When a branch is used, user essentially wants the latets
# 'snapshot' from the specified branch (version)
CURRENT_BRANCH=`git branch | sed -n '/\* /s///p'`
if ["$CURRENT_BRANCH" != "(no branch)"]
then
  git pull
fi

bundle update
berks update
rm -rf /opt/chef-run/cookbooks
berks vendor /opt/chef-run/cookbooks

# now that we have the right commit of repositories checked out, we can provision
cd /opt/chef-run
./geoshape-provision.sh silent

# restore stored directory
popd


scripts/geoshape-install.sh
#!/bin/bash
# exit if anything returns failure
set -e

#-- parse parameters to see if a geoshape version and/or the parameter 'vagrant' have been passed in.
#   if vagrant is passed in, any other parameter will be used as geoshape version. if only one
#   param is passed in and it is not 'vagrant', it will be used as the GEOSHAPE varions.
#   valid usage examples:
#   geoshape-install.sh release-1.2
#   geoshape-install.sh release-1.1 vagrant
#   geoshape-install.sh vagrant realease-1.1
#   geoshape-install.sh 1.x       // example of using a branch to essentially get the 1.1 'snapshot' as opposed to an actual release
#   geoshape-install.sh 0e43522   // example of using a commit id to get a 'snapshot' as opposed to an actual release


if [ "$1" = "vagrant" ];
then
  GEOSHAPE_USING_VAGRANT=true
  GEOSHAPE_VERSION="$2"
elif [ "$2" = "vagrant" ];
then
  GEOSHAPE_USING_VAGRANT=true
  GEOSHAPE_VERSION="$1"
else
  GEOSHAPE_VERSION="$1"
fi

if [ -z "$GEOSHAPE_VERSION" ];
then
  echo 'geoshape version not specified, will use latest release.'
fi

echo GEOSHAPE_USING_VAGRANT: ${GEOSHAPE_USING_VAGRANT}
echo GEOSHAPE_VERSION: ${GEOSHAPE_VERSION}


# install curl
apt-get install curl -y

# install rvm
gpg --keyserver hkp://keys.gnupg.net --recv-keys D39DC0E3
curl -L https://get.rvm.io | bash -s stable

# activate the correct rvm environment
source /usr/local/rvm/scripts/rvm

# get latest rvm
rvm get stable

# install ruby, my machine has this and things work
rvm list known
rvm install ruby-2.0.0-p353  # tyler uses 193
rvm --default use 2.0.0-p353
ruby -v # will show which version is being used


# install git
apt-get install -y git

# Pull rogue-chef-repo if it doesn't already exist on the VM.
# We do this so we can execute geoshape-install from a Vagrantfile and
# without a Vagrantfile.

cd /opt
if [ -d rogue-chef-repo ];
then
  cd rogue-chef-repo
else
  git clone https://github.com/ROGUE-JCTD/rogue-chef-repo.git
  cd rogue-chef-repo
fi

if [ -z "$GEOSHAPE_VERSION" ];
then
  # discover the latest release tag
  RELEASE_TAGS=(`git tag`)
  echo 'release tags: '
  echo "${RELEASE_TAGS[@]}"
  # sort the list of branches that had '.' in them such that index 0 is the largest one
  RELEASE_TAGS_SORTED=($(printf '%s\n' "${RELEASE_TAGS[@]}"|sort -r))
  GEOSHAPE_VERSION=${RELEASE_TAGS_SORTED[0]}
  echo '----[ discovered latest release version: '${GEOSHAPE_VERSION}
fi

git checkout ${GEOSHAPE_VERSION}

gem install bundler
bundle install

berks install
cd ..

# Setup Chef Run folder
# if dna.json is in /opt/chef-run, move it out, then run the following, then put it back
# Also remove the other dna files that ware aren't using for this setup.

if [ -f chef-run/dna.json ];
then
echo "Copying existing dna.json"
cp chef-run/dna.json ./dna-copy.json
cp -r /opt/rogue-chef-repo/solo/* chef-run/
cp dna-copy.json chef-run/dna.json
rm dna-copy.json
rm chef-run/dna_database.json
rm chef-run/dna_application.json
cd chef-run
else
echo "Using default dna.json"
mkdir chef-run
cp -r /opt/rogue-chef-repo/solo/* chef-run/
rm chef-run/dna_application.json
rm chef-run/dna_database.json
cd chef-run
# Edit dna.json to use correct FQDN… Note: update the url to your server’s url’. If there is no fully qualified domain name, you can simply remove the line from the dna file
sed -i 's/fqdn/fqdn-ignore/g' dna.json
# if vagrant is specified, add "vagrant":true
if [ "$1"  = "vagrant" ];
then
echo "Vagrant specified..."
sed -i '2 i\
\  "vagrant": true,
' dna.json
fi
fi

# Change username referenced in provision.sh to correct user if the user on the box is not ‘rogue’ Note: manually view provision.sh and change the user to rogue

chmod 755 *.sh

# to install latest rogue run this .sh file. if you remove the /var/lib/geoserver_data folder, it will download it again. To keep your data, just leave the folder as is and the script will not pull down the basic data folder
./geoshape-upgrade.sh ${GEOSHAPE_VERSION} silent
